jib-core/src/main/java/com/google/cloud/tools/jib/image/ReproducibleLayerBuilder.java

```diff
@@ -100,9 +102,23 @@ public Blob build() {
     for (LayerEntry layerEntry : layerEntries) {
       // Adds the entries to uniqueTarArchiveEntries, which makes sure all entries are unique and
       // adds parent directories for each extraction path.
-      uniqueTarArchiveEntries.add(
+      TarArchiveEntry entry =
           new TarArchiveEntry(
-              layerEntry.getSourceFile().toFile(), layerEntry.getAbsoluteExtractionPathString()));
+              layerEntry.getSourceFile().toFile(), layerEntry.getAbsoluteExtractionPathString());
+
+      FilePermissions permissions =
```

Comment:
I wonder if this should be in the LayerEntry. If someone does a getPermissions on a layer entry, it would make sense that this is what's returned.

We could do this directly in the `LayerEntry` constructor `if permissions == null`? and then 
`public Optional<FilePermissions> getPermissions()` -> `public FilePermissions getPermissions()`

Reply:
I think @TadCordle initially had that in `LayerEntry` but move it out of there to here so that you know if the permission was explicitly configured or not: https://github.com/GoogleContainerTools/jib/pull/1156#discussion_r226067367